metriken-exposition/src/parquet.rs

```diff
@@ -0,0 +1,302 @@
+use std::collections::BTreeMap;
+use std::io::Write;
+use std::sync::Arc;
+
+use arrow::array::*;
+use arrow::datatypes::*;
+use arrow::error::ArrowError;
+use parquet::arrow::ArrowWriter;
+use parquet::basic::{Compression, ZstdLevel};
+use parquet::errors::ParquetError;
+use parquet::file::properties::WriterProperties;
+use parquet::format::FileMetaData;
+
+use crate::snapshot::{HashedSnapshot, Snapshot};
+use crate::HistogramSnapshot;
+
+/// Options for `ParquetWriter` controlling the output parquet file.
+pub struct ParquetOptions {
+    /// Supported compression types are None or Zstd at specified level
+    compression: Compression,
+    /// Number of rows cached in memory before being written as a `RecordBatch`
+    max_batch_size: usize,
+}
+
+impl ParquetOptions {
+    pub fn builder() -> ParquetOptionsBuilder {
+        ParquetOptionsBuilder::default()
+    }
+
+    pub fn compression(&self) -> Compression {
+        self.compression
+    }
+
+    pub fn max_batch_size(&self) -> usize {
+        self.max_batch_size
+    }
+}
+
+impl Default for ParquetOptions {
+    fn default() -> Self {
+        Self {
+            compression: Compression::UNCOMPRESSED,
+            max_batch_size: 1024 * 1024,
+        }
+    }
+}
+
+/// Used to build `ParquetOptions`.
+#[derive(Default)]
+pub struct ParquetOptionsBuilder {
+    options: ParquetOptions,
+}
+
+impl ParquetOptionsBuilder {
+    pub fn build(self) -> ParquetOptions {
+        self.options
+    }
+
+    pub fn compression_level(mut self, level: i32) -> Result<Self, ParquetError> {
+        let compression = Compression::ZSTD(ZstdLevel::try_new(level)?);
+        self.options.compression = compression;
+        Ok(self)
+    }
+
+    pub fn max_batch_size(mut self, batch_size: usize) -> Self {
+        self.options.max_batch_size = batch_size;
+        self
+    }
+}
```

Comment:
I suggest not having the `ParquetOptions`/`ParquetBuilder` split and instead just having `ParquetOptions` be the builder. It is usually not necessary to have a separate builder for options since they are generally used immediately after construction.

I have also derived `Debug` and `Clone` in the suggestion below as they should generally be derived wherever possible.
```suggestion
/// Options for `ParquetWriter` controlling the output parquet file.
#[derive(Clone, Debug)]
pub struct ParquetOptions {
    /// Supported compression types are None or Zstd at specified level
    compression: Compression,
    /// Number of rows cached in memory before being written as a `RecordBatch`
    max_batch_size: usize,
}

impl ParquetOptions {
    pub fn new() -> Self {
        Self::default()
    }

    pub fn compression_level(mut self, level: i32) -> Result<Self, ParquetError> {
        let compression = Compression::ZSTD(ZstdLevel::try_new(level)?);
        self.compression = compression;
        Ok(self)
    }

    pub fn max_batch_size(mut self, batch_size: usize) -> Self {
        self.max_batch_size = batch_size;
        self
    }
}

impl Default for ParquetOptions {
    fn default() -> Self {
        Self {
            compression: Compression::UNCOMPRESSED,
            max_batch_size: 1024 * 1024,
        }
    }
}
```

Reply:
Accepted, thanks